lib/cdo/png_utils.rb

```diff
@@ -0,0 +1,28 @@
+#
+# Utility functions for reading PNG files
+#
+module PngUtils
+  # Given the complete body of a PNG file, returns hash of source dimensions:
+  # {"x": _, "y": _}
+  def dimensions_from_png(png_body)
+    # Read the first eight bytes of the IHDR Chunk, which must always be the
+    # first chunk of the PNG file.
+    #
+    # PNG Header takes 8 bytes (0x00-0x07)
+    # IHDR chunk length takes 4 bytes (0x08-0x0b)
+    # IHDR chunk type code takes 4 bytes (0x0c-0x0f)
+    # IHDR chunk data begins at 0x10
```

Comment:
This is clever, but I wonder if it's better to not roll our own?  We use MiniMagick elsewhere:

```ruby
MiniMagick::Image.open(filename).dimensions
=> [2000, 1500]
```